pkg/gui/controllers/list_controller.go

```diff
@@ -106,7 +106,7 @@ func (self *ListController) HandleGotoBottom() error {
 
 func (self *ListController) HandleClick(opts gocui.ViewMouseBindingOpts) error {
 	prevSelectedLineIdx := self.context.GetList().GetSelectedLineIdx()
-	newSelectedLineIdx := opts.Y
+	newSelectedLineIdx := self.viewIndexToModelIndex(opts.Y)
```

Comment:
When clicking on a section header, the next model item is selected instead. I think that's fine. You might prefer that nothing happens, but I think it doesn't matter much.

Reply:
If you really have two consecutive section headers and you click on the first one, it will instead select the first model item under the second header. I'd still say that's acceptable; I don't expect anybody to click on section headers, except by accident.

For menus this can never happen because section headers are derived from menu items, so there can't be a section without items.

For other lists, it's up to the client code to decide whether they want to include empty sections or not. For the upstream divergence view, I'm still undecided whether I always want to include both section headers even if one is empty; I'm also considering maybe not having headers at all, but only a divider line that is only visible when both sections are non-empty.

Reply:
I think for symmetric diff headings would be great: it removes any ambiguity about what the items represent. And given that your cursor would start on the first item, it wouldn't affect keypress counts

Reply:
I'm not so sure the ambiguity is a problem. After all, you enter the view after seeing either `mybranch ↑3`, or `mybranch ↓14`, so you know exactly what to expect in these cases. It's only for the case of `mybranch ↑4↓9` that you need some visual guidance to tell the sections apart, and for that a simple divider line is sufficient.

Reply:
My take is that, given that in all other views where we show commits, there's a logical continuity to them, it's worth spelling out what the two sections represent. Using a separator tells you that the two sets are different but using headers instead means you've only got one extra line there which imo is a very low price to pay to be explicit.

Reply:
OK, happy to try that once we have the functionality in place.